src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java

```diff
@@ -80,6 +82,29 @@ private boolean isFileNameMatching(AuditEvent event) {
                 || event.getFileName().contains(fileName));
     }
 
+    /**
+     * Is matching by treeWalker checks.
+     *
+     * @param event event
+     * @return true if it is matching
+     */
+    private boolean isTreeWalkerChecksMatching(AuditEvent event) {
+        boolean result = false;
+        final String sourceName = event.getLocalizedMessage().getSourceName();
+        try {
+            final Class clazz = Class.forName(sourceName);
+            final String superClassName = clazz.getSuperclass().getName();
+            if ("com.puppycrawl.tools.checkstyle.api.AbstractCheck".equals(superClassName)) {
+                result = true;
+            }
+            return result;
+        }
+        catch (ClassNotFoundException ex) {
+            System.out.println("an error occurred when load patch file" + ex);
```

Comment:
We should print an error. It should be logged or thrown.

Reply:
@rnveach yes, I try to thrown, and then `accept` should solve this thrown, but `accept` is override from `Filter`, and Interface `Filter`'s `accept` does not deal with any thrown.

Reply:
Sorry, I was tracking this from main project and thought this was still main project.

Once solution if we should remove the print is to throw an unchecked exception instead of a checked exception. Like to throw IllegalState or a Runtime exception.

Reply:
@HuGanghui , please do `throw new IllegalStateException("Class " + sourceName + " not found", ex);`
it is what @rnveach  meant.

Reply:
ok, done